src/buffer.rs

```diff
@@ -19,12 +19,24 @@ pub struct Cursor {
     pub line: usize,
     /// First-byte-index of glyph at cursor (will insert behind this glyph)
     pub index: usize,
+    /// Whether to associate the cursor with the run before it (false) or the run after it (true)
+    /// if placed at the boundary between two runs
+    pub affinity: bool,
 }
 
 impl Cursor {
     /// Create a new cursor
     pub const fn new(line: usize, index: usize) -> Self {
-        Self { line, index }
+        Self::new_affinity(line, index, false)
+    }
+
+    /// Create a new cursor, specifying the affinity
+    pub const fn new_affinity(line: usize, index: usize, affinity: bool) -> Self {
```

Comment:
Nit: I think `new_with_affinity` or `with_affinity` would be a more idiomatic name for this method.